### PR TITLE
Migrate to null safety

### PR DESCRIPTION
**Pull request by robbecker-wf**
## Summary

Migrate to null safety by running the migration tool and tweaking the changes. I also swapped in the lints package for the old pedantic package.

## Testing

- Globally activate this branch locally and use it to serve various projects, load the apps

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/command_runner.dart [hunk 1]
```diff
@@ -11,7 +11,7 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
+import 'dart:async';
```
Comment (robbecker-wf) [inline lib/src/command_runner.dart:14]: Needed for `FutureOr`

File: lib/src/sse_proxy_handler.dart [hunk 2]
```diff
@@ -34,14 +34,14 @@ String _sseHeaders(String origin) => 'HTTP/1.1 200 OK\r\n'
 /// simply forwards data back and forth between clients and the actual server.
 class SseProxyHandler {
   final _httpClient = http.Client();
-  shelf.Handler _incomingMessageProxyHandler;
-  final String _proxyName;
+  shelf.Handler? _incomingMessageProxyHandler;
```
Comment (evanweible-wf) [inline lib/src/sse_proxy_handler.dart:37]: #nit we could probably make this `late` instead of nullable

File: test/chromedriver_utils.dart [hunk 3]
```diff
@@ -51,7 +51,7 @@ Future<wd.WebDriver> createWebDriver() async {
       spec: wd.WebDriverSpec.JsonWire,
       desired: capabilities,
       uri: Uri.parse(
-          'http://127.0.0.1:$chromeDriverPort/$chromeDriverUrlBase/'));
+          'http://localhost:$chromeDriverPort/$chromeDriverUrlBase/'));
```
Comment (evanweible-wf) [inline test/chromedriver_utils.dart:54]: just curious, why was this change needed?

Comment (robbecker-wf) [reply test/chromedriver_utils.dart:54]: Looks like github actions refuses connections on 127.0.0.1 https://github.com/Workiva/webdev_proxy/actions/runs/4537255878/jobs/7994904946
```
00:00 +0 -1: test/sse_proxy_handler_test.dart: Can round trip messages [E]
  SocketException: Connection refused (OS Error: Connection refused, errno = 111), address = 127.0.0.1, port = 37522
  dart:_http                                                        _HttpClient.postUrl
  package:webdriver/src/request/async_io_request_client.dart 30:36  AsyncIoRequestClient.sendRaw
  ===== asynchronous gap ===========================
  dart:_http                                                        _HttpClient.postUrl
  package:webdriver/src/request/async_io_request_client.dart 30:36  AsyncIoRequestClient.sendRaw
  ===== asynchronous gap ===========================
  dart:async                                                        _CustomZone.registerUnaryCallback
  package:webdriver/src/request/async_io_request_client.dart 22:5   AsyncIoRequestClient.sendRaw
  package:webdriver/src/common/request_client.dart 96:39            AsyncRequestClient.send
  package:webdriver/async_core.dart 62:32                           createDriver
  package:webdriver/async_io.dart 41:10                             createDriver
  test/chromedriver_utils.dart 50:30                                createWebDriver
  ===== asynchronous gap ===========================
  dart:async                                                        _CustomZone.registerUnaryCallback
  test/chromedriver_utils.dart 40:[21](https://github.com/Workiva/webdev_proxy/actions/runs/4537255878/jobs/7994904946#step:7:22)                                createWebDriver
  test/sse_proxy_handler_test.dart 47:[23](https://github.com/Workiva/webdev_proxy/actions/runs/4537255878/jobs/7994904946#step:7:24)                            main.<fn>
  ===== asynchronous gap ===========================
  dart:async                                                        _CustomZone.registerUnaryCallback
  test/sse_proxy_handler_test.dart 36:45                            main.<fn>
```

File: lib/src/sse_proxy_handler.dart (hunk 2 above)
Comment (robbecker-wf) [reply lib/src/sse_proxy_handler.dart]: done

**Review by evanweible-wf — APPROVED** [review]

File: test/chromedriver_utils.dart (hunk 3 above)
Comment (evanweible-wf) [reply test/chromedriver_utils.dart:58]: So did we eventually determine that this was a red herring?

Comment (robbecker-wf) [reply test/chromedriver_utils.dart:58]: No turns out that this seems to still cause issues unless it's changed to localhost

**Review by evanweible-wf — APPROVED** [review]

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM
